static/third/bootstrap-typeahead/typeahead.js

```diff
@@ -49,14 +49,24 @@
  *   Our custom changes include all mentions of this.header, some CSS changes
  *   in compose.scss and splitting $container out of $menu so we can insert
  *   additional HTML before $menu.
- *
+ * 
  * 4. Navbar changes:
  *
  *  Typically, typeahead hotkey actions are independent of other application
  *  actions, however, the navbar is one case where we want to try and do more
  *  than a simple typeahead action with a single hotkey press, and so we've
  *  been forced to make a custom modification, see inline comment at `Esc`
  *  keyup for more details.
+ *
+ * 5. Event swallowing for Enter and Tab:
+ * 
+ *   This allows users to select items from the typeahead, without triggering
+ *   other events such as form submissions (eg when using the inline topic
+ *   edit form). This custom change does not effect selecting of items from
+ *   the typeahead in the search bar (doing so would still trigger search).
+ * 
+ *   Our custom changes are in the 'move' method, (also indicated by a short
+ *   comment there as well).
```

Comment:
This reminds me of a related bug that I fixed a while back: https://github.com/zulip/zulip/commit/1826bac1b1d7e98825058167a73a8065eaef1bf9

I'm wondering if there are other places in the codebase where people are doing bad things in `updater` methods.

I think this is mostly orthogonal to your change, so feel free to ignore. :)

Reply:
I guess the only action item here is to do a quick sanity check on the Default Streams UI in settings.